### PR TITLE
[nibeheatpump] Updated NibeGW readme

### PR DESCRIPTION
**Pull request by paulianttila**
Added build information for ProDiNo ESP32 Ethernet v1 board.

Signed-off-by: Pauli Anttila pauli.anttila@gmail.com

### CODE REVIEW

File: bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/README.md [hunk 1]
```diff
@@ -85,7 +85,30 @@ Enable support for HARDWARE_SERIAL:
 
 ##  ProDiNo ESP32 Ethernet v1 
 
-Todo
+NibeGW default settings are valid for ProDiNo ESP32 Ethernet v1 board.
+Dynamic configuration is enabled by default.
+
+### Config.h:
+
+```
```
Comment (jlaur) [inline bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/README.md:93]: Perhaps syntax highlighting could be better if language is provided?
```suggestion
```c
```

File: bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/README.md [hunk 2]
```diff
@@ -85,7 +85,30 @@ Enable support for HARDWARE_SERIAL:
...
+//#define PRODINO_BOARD
+#define PRODINO_BOARD_ESP32
+//#define TRANSPORT_ETH_ENC28J6A0
+```
+
+### NibeGW.h:
+
+```
```
Comment (jlaur) [inline bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/README.md:101]:
```suggestion
```c
```

File: bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/README.md [hunk 3]
```diff
@@ -95,15 +118,18 @@ Todo
...
 ```
 #define ENABLE_DEBUG
 #define VERBOSE_LEVEL 5
 #define ENABLE_REMOTE_DEBUG     // Remote debug is available in telnet port 23
 ```
 
-You can connect to NibeGW with any telnet client. You can also set some options via telnet. With 'h' you get a menue with all available options:
+You can connect to NibeGW with any telnet client.
+You can also set some options via telnet.
+With 'h' you get a menue with all available options:
```
Comment (jlaur) [inline bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/README.md:132]:
```suggestion
With 'h' you get a menu with all available options:
```

File: bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/README.md [hunk 4]
```diff
@@ -123,3 +149,24 @@ On the target IP you can see the receiving udp messages with netcat (if you chan
 nc -lu 9999 | hexdump -C
 ```
 
+## Dynamic configuration
```
Comment (jlaur) [inline bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/README.md:152]:
```suggestion
## Dynamic Configuration
```

File: bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/README.md [hunk 5]
```diff
@@ -123,3 +149,24 @@ On the target IP you can see the receiving udp messages with netcat (if you chan
...
+
+When dynamic configuration is enabled (only ESP32 boards), NibeGW can be configured via WiFi connection.
```
Comment (jlaur) [inline bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/README.md:154]:
```suggestion
When dynamic configuration is enabled (only ESP32 boards), NibeGW can be configured via Wi-Fi connection.
```

File: bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/README.md [hunk 6]
```diff
@@ -123,3 +149,24 @@ On the target IP you can see the receiving udp messages with netcat (if you chan
...
+Also OTA firmware update is supported.
+
+Following libraries are required
```
Comment (jlaur) [inline bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/README.md:157]:
```suggestion
The following libraries are required:
```

File: bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/README.md [hunk 7]
```diff
@@ -123,3 +149,24 @@ On the target IP you can see the receiving udp messages with netcat (if you chan
...
+ * Bleeper (tested with version 1.1.0)
+ * ElegantOTA (tested with version 2.2.9)
+
+Dynamic configuration mode is loaded if input 0 is ON during boot.
+When dynamic configuration mode is activated, login to the 'Bleeper' WiFi Access point.
```
Comment (jlaur) [inline bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/README.md:162]:
```suggestion
When dynamic configuration mode is activated, login to the 'Bleeper' Wi-Fi access point.
```

File: bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/README.md [hunk 8]
```diff
@@ -123,3 +149,24 @@ On the target IP you can see the receiving udp messages with netcat (if you chan
...
+Configuration page is available on port 80.
+OTA update page on port 8080.
+
+### Disable dynamic configuration
+
+Dynamic configuration can be disabled by commenting out following line from config.h file.
+
+```
```
Comment (jlaur) [inline bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/README.md:170]:
```suggestion
```c
```

File: bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/README.md [hunk 9]
```diff
@@ -123,3 +149,24 @@ On the target IP you can see the receiving udp messages with netcat (if you chan
...
+Following libraries are required
+ * Bleeper (tested with version 1.1.0)
+ * ElegantOTA (tested with version 2.2.9)
+
+Dynamic configuration mode is loaded if input 0 is ON during boot.
+When dynamic configuration mode is activated, login to the 'Bleeper' WiFi Access point.
+Configuration page is available on port 80.
+OTA update page on port 8080.
+
+### Disable dynamic configuration
```
Comment (jlaur) [inline bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/README.md:166]:
```suggestion
### Disable Dynamic Configuration
```

**Review by jlaur — CHANGES_REQUESTED** [review]
Thanks! Only some minor proof-reading comments.

**Review by jlaur — APPROVED** [review]
LGTM

**Comment by ajaysanan** [comment]
Having just started looking around github and trying to start my first binding, I'm certain I've clicked on something incorrectly and I apologize.  I really do not know what this push means that "referenced this pull request".  I hope it did not cause any issue; I had assumed everything I changed applied only to the fork I had created.

**Pull request merged**